File: Temperatures.cs
using System;
using System.Linq;
using System.IO;
using System.Text;
using System.Collections;
using System.Collections.Generic;

class Solution {
    static void Main(string[] args) {
        int n = int.Parse(Console.ReadLine()); // the number of temperatures to analyse
        string temps = Console.ReadLine(); // the n temperatures expressed as integers ranging from -273 to 5526

        int lowestTemp = 0;
        
        if (n > 0) {
            int marker = 0;
            int[] temperatures = new int[n];
            
            string[] tempContainer = new string[n];
            tempContainer = temps.Split(' ');
            for (int i = 0; i < n; i++) {
                temperatures[i] = int.Parse(tempContainer[i]);
            }
            
            
            int currTemp = 0;
            
            for (int i = 0; i < n; i++)
            {
                currTemp = temperatures[i];                
                if (lowestTemp == 0) {
                    lowestTemp = currTemp;
                } else {
                    if (Math.Abs(currTemp) < Math.Abs(lowestTemp)) lowestTemp = currTemp;
                    else if (Math.Abs(currTemp) == Math.Abs(lowestTemp)) {
                        if (currTemp != lowestTemp) lowestTemp = Math.Abs(lowestTemp);
                    }
                }
            }
        }
        
        Console.WriteLine(lowestTemp);
    }
}

File: Temperatures.go
package main

import "fmt"
import "os"
import "bufio"
import "strings"
import "strconv"
import "math"

func main() {
    scanner := bufio.NewScanner(os.Stdin)
    scanner.Buffer(make([]byte, 1000000), 1000000)

    // n: the number of temperatures to analyse
    var n int
    scanner.Scan()
    fmt.Sscan(scanner.Text(),&n)
    
    scanner.Scan()
	inputs := strings.Split(scanner.Text()," ")
	var lowestTemp int64 = 0
    for i := 0; i < n; i++ {
        // t: a temperature expressed as an integer ranging from -273 to 5526
        t,_ := strconv.ParseInt(inputs[i],10,32)
		
		if (i == 0) {
			lowestTemp = t
		} else {
			if (math.Abs(float64(t)) < math.Abs(float64(lowestTemp))) {
				lowestTemp = t
			} else if (math.Abs(float64(t)) == math.Abs(float64(lowestTemp))) {
				if (t != lowestTemp) {
				    lowestTemp = int64(math.Abs(float64(lowestTemp)))
				}
			}
		}
    }
    
    fmt.Println(lowestTemp)
}
